### PR TITLE
First draft of ASS/SSA to webVTT live conversion

### PR DESCRIPTION
**Pull request by rmikhael**
What is supported is detailed in https://ellation.atlassian.net/wiki/spaces/ELVIS/pages/284655618/ASS+SSA+to+webVTT+in+nginx-vod-module

Most current players don't honor styles, so all fonts/colors/shadows/outlines/strikeouts are ignored. Only timing/text/positioning/alignment will be followed, in addition to inline tags.

Keeping all changes now in a single .c file, to make debugging and changes easier. Code need a lot of refactoring and optimization, as well as splitting the code into parsing functions in one file with build functions in another.

A small bug is found for cases of live conversion from srt source, where the existing nginx-vod-module doesn't set the clip_to value, leading to a logged warning everytime a vtt segment is requested. I didn't want to make this fix, to avoid changing anything for non-ass sources. Just FYI

### CODE REVIEW

File: ngx_http_vod_module.c [hunk 1]
```diff
@@ -2382,6 +2382,7 @@ ngx_http_vod_update_track_timescale(
 	part = &track->frames;
 	cur_frame = part->first_frame;
 	last_frame = part->last_frame;
+
```
Comment (ssarkarVRV) [inline ngx_http_vod_module.c]: new line ??

File: vod/subtitle/ass_format.c [hunk 2]
```diff
@@ -39,77 +39,131 @@
...
+#define FIXED_WEBVTT_CLASS_UNDER_WIDTH  64
+#define FIXED_WEBVTT_CLASS_NOSTRIKE    "STYLE\r\n::cue(.nostrike) {\r\ntext-decoration: none;\r\n}\r\n\r\n"
+#define FIXED_WEBVTT_CLASS_NOSTRIKE_WIDTH  56
+#define FIXED_WEBVTT_CLASS_STRIKE      "STYLE\r\n::cue(.strike) {\r\ntext-decoration: solid line-through;\r\n}\r\n\r\n"
+#define FIXED_WEBVTT_CLASS_STRIKE_WIDTH  68
 
 #define VALIGN_SUB 0
 #define VALIGN_CENTER 8
 #define VALIGN_TOP 4
-#define HALIGN_LEFT 1
```
Comment (ssarkarVRV) [inline vod/subtitle/ass_format.c]: why were these removed ?

File: ngx_http_vod_module.c (hunk 1 above)
Comment (rmikhael) [inline ngx_http_vod_module.c]: if clip_to is not set to UINT_MAX, we get a warning here for srt sources.

File: vod/subtitle/ass_format.c [hunk 3]
```diff
@@ -1384,83 +1438,330 @@ static ass_track_t *parse_memory(char *buf, int len, request_context_t* request_
...
+                    default: {
+                        // replace the string with its equivalent in target string
+                        vod_memcpy(textp[chunkidx] + dstidx, tag_replacement_strings[tagidx], tag_string_len[tagidx][1]);
+                        dstidx += tag_string_len[tagidx][1]; //tag got written to output
+                    }
                 }
 
-                if (textp[chunkidx-1] == (buf + charidx) && (tagidx > 1))
-                {   //consecutive tag sequences, or one at the very start of the line text
 #ifdef  TEMP_VERBOSITY
```
Comment (ssarkarVRV) [inline vod/subtitle/ass_format.c]: what is TEMP_VERBOSITY ?

Comment (ssarkarVRV) [reply vod/subtitle/ass_format.c]: unless required; it should be removed.

File: vod/subtitle/ass_format.c (hunk 2 above)
Comment (rmikhael) [reply vod/subtitle/ass_format.c]: Taken from libass, for rendering code that is not used here. I only needed the parsing code.

File: ngx_http_vod_module.c (hunk 1 above)
Comment (rmikhael) [reply ngx_http_vod_module.c]: removing...

File: vod/subtitle/ass_format.c (hunk 3 above)
Comment (rmikhael) [reply vod/subtitle/ass_format.c]: TEMP_VERBOSITY is debugging code. Disabled by default, but makes life easier when developing. I could remove it but it will spare time next time I am adding a feature.
